src/main/java/org/rumbledb/expressions/CommaExpression.java

```diff
@@ -60,24 +60,21 @@ public CommaExpression(ExceptionMetadata metadata) {
         return result;
     }
 
-    private boolean bypassCurrentExpressionForExecutionModeOperations() {
-        return this.expressions.size() == 1;
-    }
-
     @Override
     public void initHighestExecutionMode(VisitorConfig visitorConfig) {
-        if (bypassCurrentExpressionForExecutionModeOperations()) {
-            return;
-        }
-        super.initHighestExecutionMode(visitorConfig);
-    }
+        ExecutionMode currentMode = ExecutionMode.LOCAL;
```

Comment:
This can be simplified and made more readable:

if(this.expressions.isEmpty())
{
  this.highestExecutionMode = ExecutionMode.LOCAL;
  return;
}

for(Expression expression : this.expressions)
{
    if (expression.getHighestExecutionMode(visitorConfig).isLocal())
    {
          this.highestExecutionMode = ExecutionMode.LOCAL;
          return;
    }
}

this.highestExecutionMode = ExecutionMode.RDD;

Reply:
Done.